CMakeLists.txt

```diff
@@ -1,7 +1,6 @@
 cmake_minimum_required(VERSION 3.1)
 project(qp_solver_collection)
 
-add_compile_options(-std=c++17)
```

Comment:
Why remove this and separate it into the following two?
- https://github.com/isri-aist/QpSolverCollection/pull/5/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR30
- https://github.com/isri-aist/QpSolverCollection/pull/5/files#diff-148715d6ea0c0ea0a346af3f6bd610d010d490eca35ac6a9b408748f7ca9e3f4R6

Reply:
The `cxx_std_17` meta-feature of `target_compile_features` is only available from CMake 3.8 ( https://cmake.org/cmake/help/latest/release/3.8.html?highlight=cxx_std_17 ), while the current cmake_minimum_required is 3.1 . To avoid breaking any possible existing use of this library with CMake > 3.1 and < 3.8, I continued to use `add_compile_options(-std=c++17)` when using catkin, and I switched to  `cxx_std_17` when not using catkin. If we want to get rid of this duplication, we can bump cmake_minimum_required to 3.8 .

Reply:
The `cxx_std_17` meta-feature of `target_compile_features` is only available from CMake 3.8 ( https://cmake.org/cmake/help/latest/release/3.8.html?highlight=cxx_std_17 ), while the current cmake_minimum_required is 3.1 . To avoid breaking any possible existing use of this library with CMake > 3.1 and < 3.8, I continued to use `add_compile_options(-std=c++17)` when using catkin, and I switched to  `cxx_std_17` when not using catkin. If we want to get rid of this duplication, we can bump cmake_minimum_required to 3.8 .

Reply:
> If we want to get rid of this duplication, we can bump cmake_minimum_required to 3.8 .

Thanks for the explanation. I updated in https://github.com/isri-aist/QpSolverCollection/pull/5/commits/00724278f4284f93a4084f1caf06957f3c730212